src/Razor/Microsoft.AspNetCore.Razor.Language/src/BoundAttributeDescriptorComparer.cs

```diff
@@ -67,7 +67,11 @@ public virtual int GetHashCode(BoundAttributeDescriptor descriptor)
                 }
             }
 
-            hash.Add(descriptor.Metadata.Count);
+            foreach (var metadata in descriptor.Metadata)
```

Comment:
Ahh, I see we were already comparing them in Equals but not in GetHashCode. This looks like straight up goodness